RELEASING/changelog.py

```diff
@@ -0,0 +1,255 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import os
+import re
+import urllib
+from dataclasses import dataclass
+from time import sleep
+from typing import Dict, List, Union
+from urllib import request
+
+import click
+
+try:
+    import click
+except ModuleNotFoundError:
+    exit("Click is a required dependency for this script")
+
+
+@dataclass
+class GitLog:
+    """
+    Represents a git log entry
+    """
+
+    sha: str
+    author: str
+    time: str
+    message: str
+    pr_number: Union[int, None] = None
+
+    def __eq__(self, other: "GitLog") -> bool:
+        """ A log entry is considered equal if it has the same PR number """
+        if isinstance(other, self.__class__):
+            return other.pr_number == self.pr_number
+        return False
+
+    def __repr__(self):
+        return f"[{self.pr_number}]: {self.message} {self.time} {self.author}"
+
+
+class GitChangeLog:
+    """
+    Helper class to output a list of logs entries on a superset changelog format
+
+    We want to map a git author to a github login, for that we call github's API
+    """
+
+    def __init__(self, logs: List[GitLog]) -> None:
+        self._logs = logs
+        self._github_login_cache = {}
```

Comment:
We can preset the cache here, for example with the currently more frequent committers